feat(button): add fullwidth prop - FE-2803 (#2970)
Sage/carbon: src/components/button/button.stories.mdx

  </Story>
</Preview>

> Dashed buttons can be `fullWidth` if their size is `large`.

Reviewer: Any size button can be full width

Reviewer 2: Good spot. I'll get this changed.

  </Story>
</Preview>

## Dashed Full Width Button

> Note, only button with props of `{ size: 'large' }` and `{ buttonType: 'dashed' }` will accept/work with `fullWidth` prop.

Reviewer 3: ```suggestion
Note, only large dashed (`size='large' buttonType='dashed'`) buttons can be `fullWidth`.
```

Author: updated.
